bioptim/interfaces/ipopt_interface.py

```diff
@@ -227,6 +227,23 @@ def get_x_and_u_at_idx(_penalty, _idx):
                 ocp = self.ocp
                 _x = vertcat(ocp.nlp[_penalty.phase_pre_idx].X[-1], ocp.nlp[_penalty.phase_post_idx].X[0][:, 0])
                 _u = vertcat(ocp.nlp[_penalty.phase_pre_idx].U[-1], ocp.nlp[_penalty.phase_post_idx].U[0])
+            elif _penalty.multinode_constraint:
+                ocp = self.ocp
+                _x = vertcat(
+                    ocp.nlp[_penalty.phase_first_idx].X[_penalty.node_idx[0]],
+                    ocp.nlp[_penalty.phase_second_idx].X[_penalty.node_idx[1]][:, 0],
+                )
+                # Make an exception to the fact that U is not available for the last node
+                mod0 = 0
```

Comment:
if it is only used for u, please specify it in the name (`mod_u`)

Reply:
done